packages/suite/src/components/suite/layouts/SuiteLayout/PageHeader/GlobalSendReceive/GlobalSendReceive.tsx

```diff
@@ -13,36 +14,62 @@ import { GlobalReceiveModal } from './GlobalReceiveModal';
 import { GlobalSendModal } from './GlobalSendModal';
 import { GlobalSendReceiveButtons } from './GlobalSendReceiveButtons';
 
+type SendModalEventOptions = Extract<
+    SuiteAnalyticsEvent,
+    { type: EventType.DashboardSendModalOptions }
+>['payload']['option'];
+
+type ReceiveModalEventOptions = Extract<
+    SuiteAnalyticsEvent,
+    { type: EventType.DashboardReceiveModalOptions }
+>['payload']['option'];
+
+const getDashboardParamModal = (param: unknown): GlobalSendReceiveType => {
+    if (typeof param === 'object' && param !== null && 'modal' in param) {
+        const { modal } = param as Record<string, unknown>;
+
+        return modal === 'send' || modal === 'receive' ? modal : null;
+    }
+
+    return null;
+};
```

Comment:
What about using `zod` or some schema library for this? It would be possible to define specific schema, infer type and validate against it. E.g.:

```ts
import { z } from 'zod'

const dashboardParams = z.object({
   modal: z.enum(['send', 'receive']).nullable()
})

type DashboardParams = z.infer<typeof dashboardParams>
type GlobalSendReceiveType = DashboardParams['modal']

function getDashboardParamModal(params: unknown) {
const result = dashboardParams.safeParse(router.params)

return result.success ? result.data.modal : null
}
```